# This config file is used by ros2_control
/**:
  controller_manager:
    ros__parameters:
      update_rate: 500  # Hz
      # is_async: true
          
      joint_state_broadcaster:
        type: joint_state_broadcaster/JointStateBroadcaster
      imu_sensor_broadcaster:
        type: imu_sensor_broadcaster/IMUSensorBroadcaster
      hotdog_controller:
        type: hotdog_locomotion/NewRobotController
      effort_controller:
        type: effort_controllers/JointGroupEffortController
      
  # TODO: imcomplete
  imu_sensor_broadcaster:
    ros__parameters:
      update_rate: 500  # Hz
      sensor_name: trunk_imu
      frame_id:  trunk_imu
  joint_state_broadcaster:
    ros__parameters:
      update_rate: 100  # Hz

  hotdog_controller:
    ros__parameters:
      update_rate: 500  # Hz
      joints:

        - Right-qian-joint-1
        - Right-qian-joint-2
        - Right-qian-joint-3

        - Left-qian-joint-1
        - Left-qian-joint-2
        - Left-qian-joint-3

        - Right-hou-joint-1
        - Right-hou-joint-2
        - Right-hou-joint-3

        - Left-hou-joint-1
        - Left-hou-joint-2
        - Left-hou-joint-3
      command_interfaces:
        - position
        - velocity
        - kp
        - kd
        - effort
      state_interfaces:
        - position
        - velocity
        - effort
      # wheel_joint_name:
      #   - joint_frontleft_leg_4
      #   - joint_frontright_leg_4
      #   - joint_rearleft_leg_4
      #   - joint_rearright_leg_4
      # base_name: base_link  
      sensors: 
        - trunk_imu


  effort_controller:
    ros__parameters:
      joints:

        - Right-qian-joint-1
        - Right-qian-joint-2
        - Right-qian-joint-3

        - Right-hou-joint-1
        - Right-hou-joint-2
        - Right-hou-joint-3
                
        - Left-qian-joint-1
        - Left-qian-joint-2
        - Left-qian-joint-3

        - Left-hou-joint-1
        - Left-hou-joint-2
        - Left-hou-joint-3

